P2PEnabledConference.js

```diff
@@ -0,0 +1,899 @@
+ * available in case there are only 2 participants in the room. The JVB
+ * connection will be kept alive and it will be reused if the 3rd participant
+ * joins.
+ *
+ * When the conference is being switched from one mode to another the local
+ * tracks are detached from inactive session (through JingleSessionPC). It means
+ * that locally those tracks are removed from the underlying PeerConnection, but
+ * are still signalled to the remote participants. No data is being sent for
+ * those tracks.
+ * As for the remote tracks those are replaced by generating fake "remote track
+ * added/removed" events.
+ */
+export default class P2PEnabledConference extends JitsiConference {
+    /**
+     * Creates new <tt>P2PEnabledConference</tt>.
+     * @param options see description in {@link JitsiConference} constructor.
+     * @param {number} [options.config.backToP2PDelay=5] a delay given in
+     * seconds, before the conference switches back to P2P after the 3rd
+     * participant has left.
+     */
+    constructor(options) {
+        // Call super
+        super(options);
+        // Original this.eventEmitter.emit method, stored to skip the event
+        // filtering logic
+        this._originalEmit = this.eventEmitter.emit.bind(this.eventEmitter);
+        // Intercepts original event emitter calls to filter out some of
+        // the conference events
+        this.eventEmitter.emit = this._emitIntercept.bind(this);
+        /**
+         * Stores reference to deferred start P2P task. It's created when 3rd
+         * participant leaves the room in order to avoid ping pong effect (it
+         * could be just a page reload).
+         * @type {number|null}
+         */
+        this.deferredStartP2P = null;
+
+        const delay = parseInt(options.config.backToP2PDelay);
+        /**
+         * A delay given in seconds, before the conference switches back to P2P
+         * after the 3rd participant has left.
+         * @type {number}
+         */
+        this.backToP2PDelay = isNaN(delay) ? 5 : delay;
+        logger.info("backToP2PDelay: " + this.backToP2PDelay);
+
+        /**
+         * If set to <tt>true</tt> it means the P2P ICE is no longer connected.
+         * When <tt>false</tt> it means that P2P ICE (media) connection is up
+         * and running.
+         * @type {boolean}
+         */
+        this.isP2PConnectionInterrupted = false;
+        /**
+         * Flag set to <tt>true</tt> when P2P session has been established
+         * (ICE has been connected).
+         * @type {boolean}
+         */
+        this.p2pEstablished = false;
+        /**
+         * Fake <tt>ChatRoom</tt> passed to {@link peerToPeerSession}.
+         * @type {FakeChatRoomLayer}
+         */
+        this.peerToPeerFakeRoom = null;
+        /**
+         * A JingleSession for the direct peer to peer connection.
+         * @type {JingleSessionPC}
+         */
+        this.peerToPeerSession = null;
```

Comment:
some places uses `p2p`, others use `peerToPeer`, would be nice to keep them consistent